bgcval2/functions/globalVolMean.py

```diff
@@ -0,0 +1,109 @@
+#
+# Copyright 2015, Plymouth Marine Laboratory
+#
+# This file is part of the bgc-val library.
+#
+# bgc-val is free software: you can redistribute it and/or modify it
+# under the terms of the Revised Berkeley Software Distribution (BSD) 3-clause license.
+
+# bgc-val is distributed in the hope that it will be useful, but
+# without any warranty; without even the implied warranty of merchantability
+# or fitness for a particular purpose. See the revised BSD license for more details.
+# You should have received a copy of the revised BSD license along with bgc-val.
+# If not, see <http://opensource.org/licenses/BSD-3-Clause>.
+#
+# Address:
+# Plymouth Marine Laboratory
+# Prospect Place, The Hoe
+# Plymouth, PL1 3DH, UK
+#
+# Email:
+# ledm@pml.ac.uk
+#
+"""
+.. module:: globalVolMean 
+   :platform: Unix
+   :synopsis: This function calculates the volume weighted mean of a field.
+
+.. moduleauthor:: Lee de Mora <ledm@pml.ac.uk>
+
+"""
+
+import numpy as np
+from ..bgcvaltools.dataset import dataset
+import os, sys
+import errno
+
+#from bgcvaltools.bgcvalpython import Area
+
+tmask     = 0
+pvol     = 0
+loadedArea = False
+
+def loadDataMask(gridfn):
+    """
+    Load data mask, heopfully only once!
+    """
+    global loadedArea
+    global tmask
+    global pvol
+    if not gridfn or not os.path.exists(gridfn):
+        raise FileNotFoundError(errno.ENOENT, os.strerror(errno.ENOENT), gridfn)
+
+    nc = dataset(gridfn, 'r')        
+    try:
+        pvol   = nc.variables['pvol'][:]
+        tmask = nc.variables['tmask'][:]
+    except:
+        tmask = nc.variables['tmask'][:]
+        area = nc.variables['e2t'][:] * nc.variables['e1t'][:]
+        pvol = nc.variables['e3t'][:] * area
+        pvol = np.ma.masked_where(tmask==0, pvol)
+    nc.close()
+    loadedArea = True
+
+
+def globalVolumeMean(nc, keys, **kwargs):
+    """
+    Calculate the global volume mean.
+    """
+    try:
+        areafile = kwargs['areafile']
+        print('globalVolumeMean:', areafile, kwargs)
+    except:
+        raise AssertionError("globalVolumeMean:\t Needs an `areafile` kwarg to calculate Global Volume Mean")
+
+    if isinstance(areafile, list) and len(areafile)==1:
+        areafile = areafile[0]
+    try:
+        addvalue = float(kwargs['addvalue'])
+        # In case you want to add a constant value to the data (usually Kelvin to Celcius)
+    except:
+        addvalue = 0.
+
+    try:
+        multiplyBy = float(kwargs['multiplyBy'])    # In case you want to multiply the data by a factor
+    except:
+        multiplyBy = 1.        
+        
+    if not loadedArea:
+         loadDataMask(areafile)
+            
+    temp = np.ma.array(nc.variables[keys[0]][:].squeeze())
+    temp = np.ma.masked_where((tmask==0) + (temp.mask), temp)
```

Comment:
I'd also do it like this:
```python
t1 = np.ma.masked_where(tmask==0)
if np.ma.is_masked(temp):
    t2 = temp.mask
    t1 |= t2
if np.any(t1):
    temp = np.ma.masked_where...
```

Reply:
this is the one I was worrying about on the phone

Reply:
what's so bad about this?

Reply:
nothing too bad - only if `temp` is not masked that'll fail, and it'd be good to get used to perform operations if they need to be performed ie if the cumm mask is empty don't do anything

Reply:
ignore if you think all's good, man, if we hit an issue we'll fix anyway :+1:

Reply:
Yeah, not too worried! Might wind up changing this function at some point anyway,